baler/modules/data_processing.py

```diff
@@ -158,10 +160,6 @@ def renormalize_func(norm_data: ndarray, min_list: List, range_list: List) -> nd
         ndarray: Array with the un-normalized values.
     """
     norm_data = np.array(norm_data)
-    renormalized = [
-        renormalize_std(norm_data, min_list[i], range_list[i])
-        for i in range(len(min_list))
-    ]
-    renormalized_full = [(renormalized[i][:, i]) for i in range(len(renormalized))]
-    renormalized_full = np.array(renormalized_full).T
-    return renormalized_full
+    min_list = np.array(min_list)
+    range_list = np.array(range_list)
+    return norm_data * range_list + min_list
```

Comment:
Numpy Matrix Multiplication and Addition are much more Optimized than ListWise Multiplication.
After testing on CPU/GPU, it seems to be faster and consumes less memory than older code.

Reply:
Yes I agree, and the operations looks a lot cleaner as well :-)